core/src/main/java/com/sequenceiq/cloudbreak/service/publicendpoint/GatewayPublicEndpointManagementService.java

```diff
@@ -276,4 +287,16 @@ private String getEndpointNameForStack(Stack stack) {
         return loadBalancerPersistenceService.findByStackId(stack.getId()).stream()
             .map(LoadBalancer::getEndpoint).collect(Collectors.toSet());
     }
+
+    private Optional<LoadBalancer> selectLoadBalancer(Set<LoadBalancer> loadBalancers) {
+        Optional<LoadBalancer> loadBalancerOptional = loadBalancers.stream()
+            .filter(lb -> lb.getType() == LoadBalancerType.PUBLIC)
+            .findFirst();
+        if (loadBalancerOptional.isEmpty()) {
+            loadBalancerOptional = loadBalancers.stream()
+                .filter(lb -> lb.getType() == LoadBalancerType.PRIVATE)
+                .findFirst();
+        }
+        return loadBalancerOptional;
+    }
```

Comment:
this method looks like a duplication except the order is changed. Could you generalize this?

Reply:
Moved to `LoadBalancerConfigService`.